### PR TITLE
CRUD module generator added; all templates codestyle updated;

### PR DESCRIPTION
**Pull request by UndeadBaneGitHub**

### CODE REVIEW

**Comment by UndeadBaneGitHub** [comment]
And here we go! CRUD module generator AND updating the code style of other templates is now here!

File: crud-module/index.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
+'use strict';
+var util = require('util'),
```
Comment (ilanbiala) [inline crud-module/index.js]: newline

File: crud-module/templates/client/controllers/_.client.controller.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          <%= camelizedSingularName %>Id: res._id
+        });
+      }
+
+      function errorCallback(res) {
+        vm.error = res.data.message;
+      }
+    }
+  }
+})();
```
Comment (ilanbiala) [inline crud-module/templates/client/controllers/_.client.controller.js]: newline

**Comment by ilanbiala** [comment]
@UndeadBaneGitHub can you make those changes and squash down to 1 commit? @codydaig can you take a look as well?

**Comment by UndeadBaneGitHub** [comment]
@ilanbiala Sure I will - just want all the bugs out of the way first, so that I'd not have to squash, squash, squash...

File: crud-module/templates/client/controllers/_.client.controller.js (hunk 2 above)
Comment (UndeadBaneGitHub) [inline crud-module/templates/client/controllers/_.client.controller.js]: added

**Comment by UndeadBaneGitHub** [comment]
As soon as I get 2 "LGTM"s, I will squash it

**Comment by trendzetter** [comment]
Thanks for informing me. I will try to look into it tonight.

**Comment by trendzetter** [comment]
How to I run a generator from sources in Ubuntu? If I run yo it uses the globally installed yo in /usr/local/bin/yo. Any pointers?

**Comment by UndeadBaneGitHub** [comment]
@trendzetter Uninstall existing generator, clone the PR/forked repo, then `npm install` to install all the dependencies, then `npm link` - and it will be globally available. `npm link` has to be `sudo`ed.

**Comment by trendzetter** [comment]
The module looks up to date and functional. I got 2 indent "errors":
```
modules/items/client/config/items.client.routes.js
  36:13  error  Expected indentation of 10 space characters but found 12  indent
  49:13  error  Expected indentation of 10 space characters but found 12  indent

✖ 2 problems (2 errors, 0 warnings)
```
I also had to give a name for the module while I already had given it as a parameter:
```
$ yo meanjs:crud-module item
? What is the name of the module? 
```

**Comment by UndeadBaneGitHub** [comment]
@trendzetter I will fix the indentation shortly. 

As for the name: all generators now ask for the name after being ran, ignoring the inline name. Docs should be updated real soon, I fathom.

**Comment by rof20004** [comment]
Identation warning can be ignored, I am using right now and works normally. I think the docs is the next step right ? =D.

Thanks for Identation fix.

**Comment by ilanbiala** [comment]
@UndeadBaneGitHub I think everything looks fine. Are there any more tests maybe that you can add?

**Comment by UndeadBaneGitHub** [comment]
@ilanbiala Cool. Well, I could, probably, but the thing is, the generator is quite complex. Testing for files contents is kinda useless - way too many files that are really tightly linked. So I ran out of ideas really fast - and just stopped with that single "files exist" test.

Got any suggestions?

**Comment by ilanbiala** [comment]
I guess there's nothing else that's easily testable.

**Comment by vinayakvinay** [comment]
What is the latest working version of Node with meanjs generator?

**Comment by vinayakvinay** [comment]
@ilanbiala What is the system requirement to use all the features of generator meanjs? I just want to use crud module. Please give me the version details for listed items below:
Node version ??
Npm version ??
Generator version??

**Comment by ilanbiala** [comment]
Node <=4, npm <=2, latest version of generator.

**Comment by ilanbiala** [comment]
@codydaig can you take a look as well to see if I missed anything?

**Comment by UndeadBaneGitHub** [comment]
@codydaig Yep, looking at it would be great! I can squash it anytime. So, second "LGTM" is the last thing :)

Also, I tested the generators on 5.x.x versions of node as well.

**Comment by reuf** [comment]
woohoo, livin da vida mean.js

**Comment by vinayakvinay** [comment]
@ilanbiala 
Thank you so much sir! Right now, I am on 5.x version of node and I am ready to downgrade it for super awesome mean generator. 

Can I use crud-module generator with node 4.x & latest generator version?

**Comment by UndeadBaneGitHub** [comment]
@vinayakvinay I don't think you really need to - while 5.x.x is not officially in the supported list, both generator and mean work with it. Well, at least they work both on my Windows and Debian Jessie and integration tests pass.

So, it's likely you'll be fine with 5.x.x :)

**Comment by UndeadBaneGitHub** [comment]
@codydaig @ilanbiala Squashing the commits, so if everything's good, you would be able to merge the PR

**Comment by codydaig** [comment]
LGTM. 

@UndeadBaneGitHub Sorry for the delay, but thanks for the hard work. :-D

**Comment by vinayakvinay** [comment]
@UndeadBaneGitHub @ilanbiala @codydaig 
Awesome! I will continue with latest node version. Waiting for the killer feature - crud module generator with latest mean-generator. Let us know when it is included with generator.

**Pull request merged**